### PR TITLE
Fix Whole Rest Bbox/Cursor

### LINKED ISSUE #380: Cursor misplaced for whole rest
image
This still happens in the cursorStepping branch, not sure why.
For a sample, try the first OSMD Function test.

### PR DESCRIPTION
**Pull request by sschmidTU**
Fix #380

* Fix whole rest Bounding Box (for StaffEntry, VoiceEntry, GraphicalNote), fixing Cursor position
image

* Add GraphicalNotes and VoiceEntries options to show bbox
* Demo: Rename "Show bounding box" options to be more similar to class names
image

The bbox fix isn't pretty, but that's because the bounding box given by Vexflow is completely wrong.

Test Xml with some rest variations:
[Whole_Rest_Bbox_test.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/2376420/Whole_Rest_Bbox_test.zip)

### CODE REVIEW

File: src/MusicalScore/Graphical/MusicSheetCalculator.ts [hunk 1]
```diff
@@ -161,7 +161,7 @@ export abstract class MusicSheetCalculator {
         }
         this.handleStaffEntries();
         this.calculateVerticalContainersList();
-        this.setIndecesToVerticalGraphicalContainers();
+        this.setIndicesToVerticalGraphicalContainers();
```
Comment (bneumann) [inline src/MusicalScore/Graphical/MusicSheetCalculator.ts]: Hehe ...word nerd :D

File: src/MusicalScore/Graphical/VexFlow/VexFlowStaffEntry.ts [hunk 2]
```diff
@@ -29,6 +30,12 @@ export class VexFlowStaffEntry extends GraphicalStaffEntry {
                 gve.vfStaveNote.setStave(stave);
                 gve.applyBordersFromVexflow();
                 this.PositionAndShape.RelativePosition.x = gve.vfStaveNote.getBoundingBox().x / unitInPixels;
+                const sourceNote: Note = gve.notes[0].sourceNote;
+                if (sourceNote.isRest() && sourceNote.Length.WholeValue === 1) { // whole rest
+                    this.PositionAndShape.RelativePosition.x -= 2.6; // xShift from VexFlowConverter
```
Comment (bneumann) [inline src/MusicalScore/Graphical/VexFlow/VexFlowStaffEntry.ts]: Yep we should create an EngravingRule for this.

Comment (sschmidTU) [reply src/MusicalScore/Graphical/VexFlow/VexFlowStaffEntry.ts]: But this is not really an Engraving issue, it's just that Vexflow returns a nonsense bounding box. If we create an EngravingRule, we might need to change the RelativePosition shift and the voice entry borders.
It took me a lot of trial and error to get decent values for RelativePosition.x and voice entry borders, so i hope it won't be changed so soon.

**Comment by sschmidTU** [comment]
You were joking about Vexflow, right?
This is what the bounding box looks like before this change:
image

**Comment by sschmidTU** [comment]
You were right about the EngravingRule, I created one: With Rules.wholeRestXShiftVexflow you can now set the whole rest offset.
I thought it would make random changes to the bounding boxes if you change the spacing, because the bounding box corrections were weird, but it's actually completely consistent, didn't even need to change the equation.

File: src/MusicalScore/Graphical/EngravingRules.ts [hunk 3]
```diff
@@ -230,8 +231,8 @@ export class EngravingRules {
         this.wedgeMinLength = 2.0;
         this.distanceBetweenAdjacentDynamics = 0.75;
 
-        // GraceNote Variables
-        this.tempoChangeMeasureValitidy = 4;
+        // Tempo Variables
+        this.tempoChangeMeasureValidity = 4;
```
Comment (sschmidTU) [inline src/MusicalScore/Graphical/EngravingRules.ts]: being a word nerd again, sorry ;)

**Pull request merged**